cpp/src/column/column.cu

```diff
@@ -275,6 +278,93 @@ struct create_column_from_view_vector {
 
 };
 
+struct concatenate_using_partition_map {
+  std::vector<column_view> const& views;
+  cudaStream_t stream;
+  rmm::mr::device_memory_resource* mr;
+
+  template <typename T,
+      std::enable_if_t<is_fixed_width<T>()>* = nullptr>
+  std::unique_ptr<column> operator()() {
+
+    // Compute the partition offsets
+    thrust::host_vector<size_type> offsets(views.size() + 1);
+    thrust::transform_inclusive_scan(thrust::host,
+      views.begin(), views.end(), offsets.begin() + 1,
+      [](column_view const& col) {
+        return col.size();
+      },
+      thrust::plus<size_type>{});
+    thrust::device_vector<size_type> d_offsets(offsets);
+    auto const output_size = offsets.back();
+
+    // Compute partition map
+    // NOTE instead of scattering `i` and doing a "max scan", scatter 1s
+    // for all but the first offset and do an inclusive scan
+    thrust::device_vector<size_type> d_partition_map(output_size);
```

Comment:
This will be a good initial test against the binary search method.

Keep in mind that the opportunity for the "partition map" method to really shine is when we can reuse the partition map for all of the columns (and bitmask) in a table.

Reply:
Consider writing the algorithms in such a way that the "partition_map" is an iterator to allow you to easily plug in the materialized partition map vs. the binary search partition map.